src/nonJvmMain/kotlin/uuid.kt

```diff
@@ -27,7 +30,7 @@ public actual class Uuid @Deprecated("Use `uuidOf` instead.", ReplaceWith("uuidO
         this.freeze()
     }
 
-    private companion object {
+    public companion object {
```

Comment:
Compiler was complaining abt this, believe it has to do with the way that the kotlinx serialization compiler plugin adjusts the companion object of the serialized class